### PR TITLE
v0.4.9 update for 0.216.9

### PR DESCRIPTION
**Pull request by toddlersizedsharkcage**
Patched for 0.216.9 and updated version to 0.4.9

### CODE REVIEW

**Comment by rogerwindberg** [comment]
@toddlersizedsharkcage can u provide a tutorial to compile? I tried to fix the reference errors, but no success.
My discord: @rogerwindberg

**Comment by toddlersizedsharkcage** [comment]
Azumatt has a good tutorial on youtube about how to set up a project called "Valheim Modding - Publicizing Assemblies & Set Up Code Refs"

**Comment by toddlersizedsharkcage** [comment]
> @toddlersizedsharkcage can u provide a tutorial to compile? I tried to fix the reference errors, but no success. My discord: @rogerwindberg

Azumatt has a good tutorial on youtube about how to set up a project called "Valheim Modding - Publicizing Assemblies & Set Up Code Refs"

**Comment by rogerwindberg** [comment]
> > @toddlersizedsharkcage can u provide a tutorial to compile? I tried to fix the reference errors, but no success. My discord: @rogerwindberg
> 
> Azumatt has a good tutorial on youtube about how to set up a project called "Valheim Modding - Publicizing Assemblies & Set Up Code Refs"

I followed the entire tutorial step by step, but the code is still full of errors.

**Comment by rogerwindberg** [comment]
@toddlersizedsharkcage, I downloaded your .dll and .pdb files and after many tests with all classes, I have this report:

# Rogue Class

This error occurs when trying to attack with the dagger:
image

And this one occurs when trying to use the Poison Bomb, with the dagger in hand:
image
(without the dagger in hand the poison bomb works normally)

When using the "Fade" skill, the animation occurs normally but the character is not invisible and the monsters can still see you and attack. Unfortunately this bug does not generate any errors in the log.
image

# Ranger

Ranger's wolf (`X` abilitie) does not damage enemies and does not generate an error in the log.
image

# Projectiles

Projectile skills don't work, it always throws this error:
image
e.g.

Mage - `Z` and `C` abilities
Monk - `C` abilitie
Priest - `Z` abilitie

**Comment by rogerwindberg** [comment]
I believe that the projectiles error has to do with the name `___m_rayMaskSolids` being changed to `___s_rayMaskSolids`
image

**Comment by toddlersizedsharkcage** [comment]
The patched version of the mod uses s_rayMaskSolids, so it appears you might be loading the old DLL somehow.  Valheim's latest (official release, not test branch) code uses s_rayMaskSolids and not m_rayMaskSolids.  I do however now see that I can't spawn a wolf at all with my patch, but everything else you mentioned seems to work for me.

I'm also not sure that the fade ability is what you're describing:
image

File: ObjectDBPatches.cs
Comment (toddlersizedsharkcage) [inline ObjectDBPatches.cs:1]: It appears a new overload for the GetStatusEffect method was added which was throwing Unity errors about ambiguous methods.

File: SE_BiomeOcean.cs
Comment (toddlersizedsharkcage) [inline SE_BiomeOcean.cs:1]: Possibly a typo that was fixed in the latest version.

**Comment by rogerwindberg** [comment]
I'm pretty sure I loaded the new DLL and I'm still having the same errors :/
Would anyone be interested in helping me validate where the problem is?

My discord: @rogerwindberg

**Comment by toddlersizedsharkcage** [comment]
> I'm pretty sure I loaded the new DLL and I'm still having the same errors :/ Would anyone be interested in helping me validate where the problem is?
> 
> My discord: @rogerwindberg

I tried both DLLs in by pull request and they seem to be operating normally.  I recommend going on the valheim modding discord or odinplus discord if you need help loading a DLL manually.

**Comment by rogerwindberg** [comment]
I think the projectiles could be some incompatibility with another mod, but the Rogue issues (with the exception of the Fade, @toddlersizedsharkcage you were right) still remain (no damage with the dagger or with the poison bomb if the dagger is equipped).

Only ValheimLegends loaded (and config manager to view this windows):

image

Console error log after hit the Greydwarf with a dagger (as you can see in file explorer, clean install):

image

**Comment by rogerwindberg** [comment]
Update: Update: After a lot of testing I found the mod that was conflicting and causing the projectiles not to work. The mod is [Valheim Defenders](https://www.nexusmods.com/valheim/mods/2289). The only problem now is that of the dagger, which still happens.

**Comment by toddlersizedsharkcage** [comment]
How do you equip a dagger?  I can't figure out what that means.  I'm able to throw a knife with block + attack and it does damage.

**Comment by NorgerLegacy** [comment]
for PTB with all the fixes in the pr,
Using this in ObjectDBPatches.cs appears to work. only test with beserker briefly no console errors.


 [HarmonyPatch(typeof(ObjectDB), "GetStatusEffect", new Type[] { typeof(int) })]
        public static class ObjectDBGetStatusEffect
        {
            public static void Postfix(ObjectDB __instance, string nameHash, StatusEffect __result)

**Review by rogerwindberg — APPROVED** [review]
LGTM! :)

**Comment by TorannD** [comment]
Updates accepted, going to close this and publish the new version to nexus mods and thunderstore.io - thanks for all the help updating and testing!

**Pull request closed without merging**